src/test/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/resource/UserCaseloadManagementResourceIntTest.kt

```diff
@@ -0,0 +1,331 @@
+        .headers(setAuthorisation(roles = listOf("ROLE_BANANAS")))
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `get user with caseloads not found`() {
+
+      webTestClient.get().uri("/users/dummy/caseloads")
+        .headers(setAuthorisation(roles = listOf("ROLE_MAINTAIN_ACCESS_ROLES_ADMIN")))
+        .exchange()
+        .expectStatus().isNotFound
+    }
+
+    @Test
+    fun `get user with caseloads`() {
+
+      webTestClient.get().uri("/users/CASELOAD_USER1/caseloads")
+        .headers(setAuthorisation(roles = listOf("ROLE_MAINTAIN_ACCESS_ROLES_ADMIN")))
+        .exchange()
+        .expectStatus().isOk
+        .expectBody()
+        .jsonPath("username").isEqualTo("CASELOAD_USER1")
+        .jsonPath("activeCaseload.id").isEqualTo("BXI")
+        .jsonPath("$.caseloads[?(@.id == '%s')]", "BXI").exists()
+        .jsonPath("$.caseloads[?(@.id == '%s')]", "WWI").exists()
+    }
+  }
+
+  @DisplayName("POST /users/{username}/caseloads/{caseloadId}")
+  @Nested
+  inner class AddCaseloadsByUsername {
+    @BeforeEach
+    internal fun createUsers() {
+      with(dataBuilder) {
+        generalUser()
+          .username("CASELOAD_USER1")
+          .firstName("MARK")
+          .lastName("BOWLAN")
+          .atPrison("BXI")
+          .buildAndSave()
+      }
+    }
+
+    @AfterEach
+    internal fun deleteUsers() = dataBuilder.deleteAllUsers()
+
+    @Test
+    fun `access forbidden when no authority`() {
+      webTestClient.post().uri("/users/CASELOAD_USER1/caseloads/LEI")
+        .exchange()
+        .expectStatus().isUnauthorized
+    }
+
+    @Test
+    fun `access forbidden when no role`() {
+
+      webTestClient.post().uri("/users/CASELOAD_USER1/caseloads/LEI")
+        .headers(setAuthorisation(roles = listOf()))
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `add caseload to user forbidden with wrong role`() {
+
+      webTestClient.post().uri("/users/CASELOAD_USER1/caseloads/LEI")
+        .headers(setAuthorisation(roles = listOf("ROLE_BANANAS")))
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `add caseloads not found`() {
+
+      webTestClient.post().uri("/users/dummy/caseloads/LEI")
+        .headers(setAuthorisation(roles = listOf("ROLE_MAINTAIN_ACCESS_ROLES_ADMIN")))
+        .exchange()
+        .expectStatus().isNotFound
+    }
+
+    @Test
+    fun `add caseload to user`() {
+
+      webTestClient.post().uri("/users/CASELOAD_USER1/caseloads/LEI")
+        .headers(setAuthorisation(roles = listOf("ROLE_MAINTAIN_ACCESS_ROLES_ADMIN")))
+        .exchange()
+        .expectStatus().isCreated
+        .expectBody()
+        .jsonPath("username").isEqualTo("CASELOAD_USER1")
+        .jsonPath("activeCaseload.id").isEqualTo("BXI")
+        .jsonPath("$.caseloads[?(@.id == '%s')]", "BXI").exists()
+        .jsonPath("$.caseloads[?(@.id == '%s')]", "LEI").exists()
+    }
```

Comment:
I always feel it is a good idea to do another GET to make sure it is definitely being persisted